cinema/serializers.py

```diff
@@ -68,13 +77,140 @@ class Meta:
+        tickets_sold = Ticket.objects.filter(movie_session=obj).count()
+        return total_capacity - tickets_sold
+
 
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = serializers.SerializerMethodField()
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+    def get_taken_places(self, obj):
+        """
+        Retrieve all taken places (row and seat) for the movie session.
+        """
+        tickets = Ticket.objects.filter(movie_session=obj)
+        return [{"row": ticket.row, "seat": ticket.seat} for ticket in tickets]
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = serializers.PrimaryKeyRelatedField(
+        queryset=MovieSession.objects.all()
+    )
+
+    class Meta:
+        model = Ticket
+        fields = (
+            "id",
+            "row",
+            "seat",
+            "movie_session",
+        )
+
+    def validate(self, data):
+        """
+        Validate ticket creation.
+        """
+        movie_session = data.get("movie_session")
+        row = data.get("row")
+        seat = data.get("seat")
+
+        # Ensure row and seat are within the cinema hall's capacity
+        cinema_hall = movie_session.cinema_hall
+        if row < 1 or row > cinema_hall.rows:
+            raise ValidationError(
+                {
+                    "row": (
+                        f"Row {row} is out of range (1 to {cinema_hall.rows})."
+                    )
+                }
+            )
+        if seat < 1 or seat > cinema_hall.seats_in_row:
+            raise ValidationError(
+                {
+                    "seat": (
+                        f"Seat {seat} is out of range (1 to "
+                        f"{cinema_hall.seats_in_row})."
+                    )
+                }
+            )
+
+        # Ensure row and seat are not already taken for this movie session
+        if Ticket.objects.filter(
+                movie_session=movie_session, row=row, seat=seat
+        ).exists():
+            raise ValidationError(
+                {
+                    "seat": (
+                        f"Row {row}, Seat {seat} is already taken "
+                        f"for this session."
+                    )
+                }
+            )
+
+        return data
+
+
+class TicketMovieListSerializer(TicketSerializer):
+    movie_session = MovieSessionListSerializer(read_only=True)
+
+    class Meta:
+        model = Ticket
+        fields = (
+            "id",
+            "row",
+            "seat",
+            "movie_session",
+        )
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, allow_empty=False)
```

Comment:
The `allow_empty=False` parameter in the `tickets` field ensures that an order cannot be created without tickets. This is a good validation step to prevent empty orders.